Apromore-ProcessMining-Collection/prom-bpmn/src/main/java/org/apromore/processmining/models/graphbased/directed/bpmn/ModelChecker.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2021 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.apromore.processmining.models.graphbased.directed.bpmn;
+
+import org.apromore.processmining.models.graphbased.directed.bpmn.elements.Activity;
+import org.apromore.processmining.models.graphbased.directed.bpmn.elements.Event;
+import org.apromore.processmining.models.graphbased.directed.bpmn.elements.Flow;
+import org.apromore.processmining.models.graphbased.directed.bpmn.elements.Gateway;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModelChecker {
+
+    private boolean allowPools;
+    private Map<Integer, List<String>> errorMap = new HashMap<>();
+
+    public static final ModelChecker MODEL_CHECKER_NO_POOLS = new ModelChecker(false);
+    public static final ModelChecker MODEL_CHECKER_ALLOW_POOLS = new ModelChecker(true);
```

Comment:
Maybe a better naming could be: MODEL_CHECKER_RESTRICTED and MODEL_CHECKER_RELAXED, each representing restricted models (e.g. pools are not allowed), or relaxed (pools are allowed). Pools is only one among many conditions, it might not be good to be part of the naming.